### PR TITLE
Reward fund account implementation

### PR DESCRIPTION
**Pull request by tanerincode**

### CODE REVIEW

File: src/Services/RewardService.php [hunk 1]
```diff
@@ -8,23 +8,25 @@
...
     /**
      * @param array $data
+     * @param string $processType
      * @return mixed
      * @throws GuzzleException
      * @throws MethodNotAllowed
      * @throws MissingParameter
      */
-    public function createReward(array $data): RewardServiceInterface {
+    public function createReward(array $data, string $processType = 'depositAccount'): RewardServiceInterface {
```
Comment (Physicliar) [inline src/Services/RewardService.php:29]: instead we should select type I think we should just get the funding account id as optional as second parameter and if funding account parameter exists we add id to the body. So we will not change the existing structure and will add optional parameter which will keep us safe.

**Review by Physicliar — APPROVED** [review]

File: src/Services/RewardService.php (hunk 1 above)
Comment (Physicliar) [reply src/Services/RewardService.php:29]: Requested changes

**Review by Physicliar — CHANGES_REQUESTED** [review]

File: src/Services/RewardService.php (hunk 1 above)
Comment (tanerincode) [reply src/Services/RewardService.php]: updated

File: src/Services/RewardService.php [hunk 2]
```diff
@@ -49,14 +49,23 @@ public function createReward(array $data): RewardServiceInterface {
...
                         ]
                     ]
                 ]
             ]
         ];
 
+        if ( $fundingAccountId ) {
+            $payload["data"]["relationships"] = [
+                "fundingAccount" => [
+                    "type" => 'fundingAccount',
```
Comment (Physicliar) [inline src/Services/RewardService.php:63]: it should be also depositAccount not funding account and can you please double quote

**Review by Physicliar — CHANGES_REQUESTED** [review]

**Review by Physicliar — APPROVED** [review]

File: src/Services/RewardService.php (hunk 2 above)
Comment (tanerincode) [reply src/Services/RewardService.php]: done

**Pull request merged**